### PR TITLE
Properly 'go to implementation' when there are abstract overrides in the hierarchy

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/19700

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz 

**Customer scenario**

Customer invokes "go to implementation" feature on a complex hierarchy.  Some overrides may not get shown to them.

**Bugs this fixes:**

Fixes #19700

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Original implementer didn't think of this case.

**How was the bug found?**

Customer report.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

File: src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs [hunk 1]
```diff
@@ -32,41 +32,48 @@ public static partial class SymbolFinder
         internal static async Task<ImmutableArray<SymbolAndProjectId>> FindOverridesAsync(
             SymbolAndProjectId symbolAndProjectId, Solution solution, IImmutableSet<Project> projects = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            // Method can only have overrides if its a virtual, abstract or override and is not
-            // sealed.
-            var symbol = symbolAndProjectId.Symbol;
+            var results = ArrayBuilder<SymbolAndProjectId>.GetInstance();
+
+            var symbol = symbolAndProjectId.Symbol?.OriginalDefinition;
             if (symbol.IsOverridable())
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs]: You are now using `?.`, but not checking for null before calling `IsOverridable()`

Comment (Pilchie) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs]: Ugh.  I see that `IsOverridable` is an extension method that allows null input, but that's just wrong.

File: src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs [hunk 2]
```diff
@@ -32,41 +32,48 @@ public static partial class SymbolFinder
...
+                var containingType = symbol.ContainingType;
                 var derivedTypes = await FindDerivedClassesAsync(
                     symbolAndProjectId.WithSymbol(containingType),
                     solution, projects, cancellationToken).ConfigureAwait(false);
 
-                var results = ArrayBuilder<SymbolAndProjectId>.GetInstance();
                 foreach (var type in derivedTypes)
                 {
-                    foreach (var m in GetMembers(type, symbol.Name))
+                    foreach (var m in type.Symbol.GetMembers(symbol.Name))
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs]: There is another call to GetMembers on L134.  Should it also be changed and `GetMembers` deleted?

**Pull request merged**